src/components/cc-email-list/cc-email-list.smart.js

```diff
@@ -0,0 +1,242 @@
+          });
+        })
+        .catch((error) => {
+          console.error(error);
+          notifyError(component, i18n('cc-email-list.secondary.action.delete.error', { address }));
+          updateSecondary(address, (secondaryAddressState) => {
+            secondaryAddressState.state = 'idle';
+          });
+        });
+    });
+
+    onEvent('cc-email-list:mark-as-primary', (address) => {
+
+      updateSecondary(address, (secondaryAddressState) => {
+        secondaryAddressState.state = 'marking-as-primary';
+      });
+
+      api.markSecondaryEmailAddressAsPrimary(address)
+        .then(() => {
+          notifySuccess(component, i18n('cc-email-list.secondary.action.mark-as-primary.success', { address }));
+
+          const primaryAddress = component.emails.value.primaryAddress.address;
+
+          updateComponent('emails', (emails) => {
+            emails.value.primaryAddress.address = address;
+          });
+          updateSecondary(address, (secondaryAddressState) => {
+            secondaryAddressState.state = 'idle';
+            secondaryAddressState.address = primaryAddress;
+          });
+        })
+        .catch((error) => {
+          console.error(error);
+          notifyError(component, i18n('cc-email-list.secondary.action.mark-as-primary.error', { address }));
+          updateSecondary(address, (secondaryAddressState) => {
+            secondaryAddressState.state = 'idle';
+          });
+        });
+    });
+  },
+});
+
+// -- API calls
+function getApi (apiConfig, signal) {
+  return {
+    fetchEmailAddresses () {
+      return Promise.all([
+        this.fetchPrimaryEmailAddress(),
+        this.fetchSecondaryEmailAddresses(),
+      ]).then(([self, secondary]) => {
+        return { self, secondary };
+      });
+    },
+
+    fetchPrimaryEmailAddress () {
+      return Promise.resolve({
+        method: 'get',
+        url: `/v2/self`,
+        headers: { Accept: 'application/json' },
+      })
+        .then(sendToApi({ apiConfig, signal }));
+    },
+
+    fetchSecondaryEmailAddresses () {
+      return getEmailAddresses()
+        .then(sendToApi({ apiConfig, signal }));
+    },
+
+    sendConfirmationEmail () {
+      return Promise.resolve({
+        method: 'get',
+        url: `/v2/self/confirmation_email`,
+        headers: { Accept: 'application/json' },
+      })
+        .then(sendToApi({ apiConfig, signal }));
+    },
+
+    addSecondaryEmailAddress (address) {
+      return addEmailAddress({ email: address }, {})
+        .then(sendToApi({ apiConfig, signal }));
+    },
+
+    deleteSecondaryEmailAddress (address) {
+      return removeEmailAddress({ email: address })
+        .then(sendToApi({ apiConfig, signal }));
+    },
+
+    markSecondaryEmailAddressAsPrimary (address) {
+      return addEmailAddress({ email: address }, {
+        // eslint-disable-next-line camelcase
+        make_primary: true,
+      })
+        .then(sendToApi({ apiConfig, signal }));
+    },
```

Comment:
We don't need the signal on those as they are update calls and we want them to go through even if there's a navigation.